doc/guides/dts/index.rst

```diff
@@ -346,44 +346,534 @@ Below is sample header content generated for this node, in
+These macros have the the format ``DT_<node>_IRQ_<index>(_<name>)``, where
+``<node>`` is the node prefix, ``<index>`` is an index that identifies the
+particular interrupt, and ``<name>`` is the identifier for the cell value (a
+number within ``interrupts = <...>``), taken from the binding.
+
+Bindings for interrupt controllers are expected to declare a cell named ``irq``
+in ``interrupt-cells``, giving the interrupt number. The ``_<name>`` suffix is
+skipped for macros generated from ``irq`` cells, which is why there's e.g. a
+``DT_VENDOR_TIMER_123_IRQ_0`` macro and no ``DT_VENDOR_TIMER_123_IRQ_0_IRQ``
+macro.
+
+If the interrupt controller in turn generates other interrupts, a multi-level
+interrupt encoding is used for the interrupt number. See
+:ref:`multi_level_interrupts` for more information.
+
+Additional macros that use names instead of indices for interrupts can be
+generated by including an ``interrupt-names`` property on the
+interrupt-generating node. For example, ``interrupt-names = "timer-a",
+"timer-b"`` gives these extra macros:
+
+.. code-block:: c
+
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_A             1
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_A_PRIORITY    5
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_B             2
+   #define DT_VENDOR_TIMER_123_IRQ_TIMER_B_PRIORITY    6
+
+Values for properties generated from ``clocks``
+-----------------------------------------------
+
+Clocks work the same as other ``phandle-array`` properties, except the
+generated macros have ``CLOCK`` in them instead of ``CLOCKS``, giving for
+example ``DT_<node>_CLOCK_CONTROLLER_0`` instead of
+``DT_<node>_CLOCKS_CONTROLLER_0``.
 
 .. note::
 
-   The above is just a short overview of common ways macro names get generated,
-   and not complete. For the nitty-gritty, see the source code in
-   :zephyr_file:`gen_defines.py <scripts/dts/gen_defines.py>`, and check the
-   output generated for some existing boards and applications.
+   This inconsistency might be fixed in the future.
+
+In addition, if the clock controller node has a ``fixed-clock`` property, it is
+expected to also have a ``clock-frequency`` property giving the frequency, and
+an additional macro is generated:
+
+.. code-block:: c
+
+   #define DT_<node>_CLOCKS_CLOCK_FREQUENCY <frequency>
+
+Macros related to SPI GPIO chip select
+--------------------------------------
+
+.. boards/arm/sensortile_box/sensortile_box.dts has a real-world example
+
+Take these devicetree nodes as an example. where the binding for
+``vendor,spi-controller`` is assumed to have ``bus: spi``, and the bindings for
+the SPI slaves are assumed to have ``on-bus: spi``:
+
+.. code-block:: none
+
+   spi {
+	compatible = "vendor,spi-controller";
+	cs-gpios = <&gpioa 1>, <&gpioa 2>;
+	spi-slave@0 {
+		compatible "vendor,foo-spi-device";
+		reg = <0>;
+	};
+	spi-slave@1 {
+		compatible "vendor,bar-spi-device";
+		reg = <1>;
+	}
+   };
+
+Here, the unit address of the SPI slaves (0 and 1) is taken as a chip select
```

Comment:
This paragraph I find confusing. what is a "chip select address"?

Reply:
changed address to number.